pkg/collector/collector.go

```diff
@@ -97,11 +96,9 @@ func NewCollector(
 		MetricsPrometheusCollector: config,
 		UtilizationServer:          utilization.Server{UtilizationServer: config.UtilizationServer},
 		startTime:                  time.Now(),
-		metrics:                    &atomic.Value{},
+		metrics:                    make(chan *cachedMetrics),
```

Comment:
Since this is a channel, would it be helpful to have a buffer? That way we can start the next caching before we are ready to send the previous one.